.github/workflows/jdl.yml

```diff
@@ -0,0 +1,247 @@
+      #----------------------------------------------------------------------
+      # Detect changes against base commit
+      #----------------------------------------------------------------------
+      - name: 'MERGE: Install base generator-jhipster'
+        continue-on-error: true
+        id: base-checkout
+        if: github.event.pull_request
+        working-directory: ${{ github.workspace }}/base/generator-jhipster
+        run: |
+          git clone ${{ github.workspace }}/generator-jhipster .
+          git checkout @~1
+          git log
+          npm uninstall -g generator-jhipster
+          npm install
+          npm install -g .
+      - name: 'MERGE: copy jdl'
+        if: steps.base-checkout.outcome == 'success'
+        working-directory: ${{ github.workspace }}/base/app
+        run: cp $JHI_JDL_SAMPLES/${{ matrix.jdl }}/*.jdl .
+      - name: 'MERGE: base project'
+        continue-on-error: true
+        id: base-app
+        if: steps.base-checkout.outcome == 'success'
+        working-directory: ${{ github.workspace }}/base/app
+        run: |
+          # docker-compose is not reproducible, so copy old config
+          ls ${{ github.workspace }}/app/docker-compose && mkdir docker-compose && cp ${{ github.workspace }}/app/docker-compose/.yo-rc.json docker-compose || true
+          jhipster jdl *.jdl --skip-jhipster-dependencies --skip-install ${{ matrix.extra-args }} --skip-git
+      - name: 'MERGE: merge diff'
+        continue-on-error: true
+        id: merge-is-equal
+        if: steps.base-app.outcome == 'success'
+        working-directory: ${{ github.workspace }}/base/app
+        run: |
+          # Copy git repository (clone without the files)
+          cp -a ${{ github.workspace }}/app/.git .
+          git log
+          # File keystore.p12 always changes
+          # File package-lock.json is not generated
+          # Verify if the generated source has changed
+          git add .
+          echo "Showing config diff changes (won't trigger builds)"
+          git -c color.ui=always diff -R --cached -- '.yo-rc.json' '**/.yo-rc.json' '.jhipster/**' '**/.jhipster/**'
+          echo "Showing source diff changes (will trigger builds)"
+          git -c color.ui=always diff -R --cached --exit-code -- . ':!.yo-rc.json' ':!**/.yo-rc.json' ':!.jhipster/**' ':!**/.jhipster/**' ':!package-lock.json' ':!**/package-lock.json' ':!**/keystore.p12' || echo "::set-output name=has-changes::true"
+      - name: 'MERGE: tests should be skipped'
+        id: tests-should-be-skipped
+        if: >
+          steps.merge-is-equal.outcome == 'success' &&
+          steps.merge-is-equal.outputs.has-changes != 'true'
+        run: echo 'Skipping tests' && echo "::set-output name=skip-tests::true"
+      #----------------------------------------------------------------------
+      # Launch tests
+      #----------------------------------------------------------------------
+      - name: 'TESTS: prepare keycloak'
```

Comment:
Can it be directly included to 03-system script?

Reply:
IMO it's too relevant to hide inside a script.